### PR TITLE
Ability to change a keyword

### LINKED ISSUE #216: Change arguments of an Action
**Is your feature request related to a problem? Please explain.**
Currently each action has access to `video, action, frame` but maybe the user would like to change a parameter.
We can use scaling for changing the radius of a circle as an example but maybe one would like to draw a line that gets longer for some frames which can currently be done hard coded with `frame` but that's not intuitive.

**Describe the solution you'd like**
One option would be:
```julia
Action(1:100, (args...; x=0, y=0)->draw_line(O, Point(x, y)); subactions=[
     SubAction(10:50, change(:x=>100))
])
```
So one can define keyword arguments that can be changed in an animated way. The change would always be from the last value to the next. I'm not sure whether Javis has internally access to the `x=0` though :thinking: 
I think we need `:x` as `x` itself is not defined.

**Describe alternatives you've considered**
If we don't have access to the default value we can maybe use:
```julia
change(:x, 0, 100)
```
or 
```julia
change(:x, 0 => 100)
```

**Additional context**
In general this should of course also work with Animations.jl such that one can write an `Animation` and just call `change(:x)` to apply that change to the variable x.

### PR DESCRIPTION
**Pull request by Wikunia**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [x] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [ ] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [x] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
- [ ] Did I check relevant tutorials that may be affected by changes in this PR?
  - We might want to change the element tutorial to use this instead of scaling. @TheCedarPrince 
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #216 

**How did you address these issues with this PR? What methods did you use?**
`change` saves the value it got by interpolation in `action.change_keywords` which is then used when calling the action.

```julia
myvideo = Video(500, 500)
javis(
    myvideo,
    [
        BackgroundAction(1:100, ground),
        Action(
            (args...; radius = 25) -> object(Point(100, 0), radius, "red"),
            Rotation(0.0, 4π);
            subactions = [
                SubAction(1:50, change(:radius, 25 => 0))
                SubAction(51:100, change(:radius, 0 => 25))
            ],
        )
    ]; pathname = "current/_change.gif"
)
```

PKvM8ov - Imgur

We might want to release this with v0.3 together but I implemented it in v0.2. Depends on how much time we need for v0.3 I suppose.

### CODE REVIEW

**Review by TheCedarPrince — COMMENTED** [review]
Great work! I really enjoy this functionality and it makes a lot of sense. However, I do have one question: why does this functionality not work as I expect it to:

```julia
using Animations
using Javis

function ground(args...)
    background("black")
    sethue("white")
end

function circ(args...; p, radius, color = "black")
    sethue(color)
    circle(p, radius, :fill)
    return p
end

function astar(args...; p, radius, color = "black")
    sethue(color)
    star(p, radius)
    return p
end

myvideo = Video(500, 500)
javis(
    myvideo,
    [
        BackgroundAction(1:100, ground),
        Action(
            (args...; radius = 25) -> circ(Point(100, 0), radius, "red"),
            Rotation(0.0, 4π);
            subactions = [
                SubAction(1:50, change(:radius, 25 => 0)),
                SubAction(morph(circ, astar)),
                SubAction(51:100, change(:radius, 0 => 25)),
                SubAction(morph(astar, circ))
            ],
        ),
    ];
    pathname = "change.gif",
)
```

Essentially what I am expecting to happen is the morph to take affect on the circle. I was expecting it as I know that Action has the associated `change_keywords` dict. Therefore, I thought the radius was being permanently mutated and could be applied to the astar function as the function definition matches that which was defined for circ. 

However, I get this error regarding morphing:

```
ERROR: LoadError: MethodError: no method matching (::Javis.var"#26#27"{Symbol,typeof(circ),typeof(astar)})(::Video, ::Action, ::SubAction, ::Int64)
Closest candidates are:
  #26(::Any, ::Any, ::Any) at /home/src/Projects/javis/src/morphs.jl:110
Stacktrace:
 [1] perform_action(::Action, ::Video, ::Int64, ::Array{Float64,2}) at /home/src/Projects/javis/src/Javis.jl:338
 [2] macro expansion at /home/src/Projects/javis/src/Javis.jl:302 [inlined]
 [3] macro expansion at /home/cedarprince/.julia/packages/Luxor/XrsCA/src/basics.jl:500 [inlined]
 [4] get_javis_frame(::Video, ::Array{Action,1}, ::Int64) at /home/src/Projects/javis/src/Javis.jl:301
 [5] macro expansion at /home/src/Projects/javis/src/Javis.jl:236 [inlined]
 [6] macro expansion at /home/cedarprince/.julia/packages/ProgressMeter/OUQkp/src/ProgressMeter.jl:732 [inlined]
 [7] javis(::Video, ::Array{Action,1}; framerate::Int64, pathname::String, liveview::Bool, tempdirectory::String) at /home/src/Projects/javis/src/Javis.jl:235
 [8] top-level scope at /home/src/Projects/javis/view.jl:22
 [9] include(::String) at ./client.jl:457
 [10] top-level scope at REPL[25]:1
in expression starting at /home/src/Projects/javis/view.jl:22
```

What am I misunderstanding? Either way, great work! Thanks Ole!

File: src/subaction_animations.jl [hunk 1]
```diff
@@ -363,3 +363,54 @@ function _follow_path(video, action, subaction, rel_frame, points; closed = clos
...
+javis(
+    myvideo,
+    [
+        BackgroundAction(1:100, ground),
+        Action(
+            (args...; radius = 25) -> object(Point(100, 0), radius, "red"),
+            Rotation(0.0, 4π);
+            subactions = [
+                SubAction(1:50, change(:radius, 25 => 0))
+                SubAction(51:100, change(:radius, 0 => 25))
```
[the remark this answers is not in the document]
Comment (Wikunia) [reply src/subaction_animations.jl:391]: I decided against it as this is just a docstring. 
The docstrings are getting very long if we always include a full example.

**Comment by Wikunia** [comment]
I haven't changed morph to also work with it. Will fix. Thanks for the review and catch

**Comment by Wikunia** [comment]
In general morph is a bit complicated as it neither a normal SubAction nor a normal Action. It kind of in between.

**Comment by Wikunia** [comment]
Okay the problem with morph is which I forgot :smile: Morph is currently an Action not a SubAction. Therefore I'll change this in v0.3 only. 
And your code can't work in general as you call the `circ` function with positional arguments but define it with kwargs. 
If you're okay with the docstring and can't break it without morph -> approve. 
Otherwise please provide a different way to break it :joy:

File: src/subaction_animations.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wikunia) [reply src/subaction_animations.jl:391]: In general docstrings for me are more for: I have forgotten the syntax and kinda wanna look what exists. But if I want something real I check out the real documentation with examples. 
For v0.3 I would also consider to just show the `Action` in some cases like `appear` without even showing the `Object`.
Does that make sense to u?

**Comment by Wikunia** [comment]
Regarding the test. Have you run `dev Javis`?

**Review by TheCedarPrince — APPROVED** [review]
LGTM - I agree with your thought on the docstrings. Ugh, I wish extended docstrings were more common. Else, I would use them all the time!

**Pull request merged**